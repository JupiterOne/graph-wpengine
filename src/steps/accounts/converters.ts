import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { Entities } from '../constants';
import { WpEngineAccount } from '../../types';

export function getAccountKey(id: string): string {
  return `wp_engine_account:${id}`;
}

export function createAccountEntity(data: WpEngineAccount) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: Entities.ACCOUNT._class,
        _type: Entities.ACCOUNT._type,
        _key: getAccountKey(data.id),
        id: data.id,
        name: data.name,
      },
    },
  });
}
