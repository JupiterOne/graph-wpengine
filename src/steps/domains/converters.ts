import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { Entities } from '../constants';
import { WpEngineDomain } from '../../types';

export function getDomainKey(id: string): string {
  return `wp_engine_domain:${id}`;
}

export function createDomainEntity(data: WpEngineDomain) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: Entities.DOMAIN._class,
        _type: Entities.DOMAIN._type,
        _key: getDomainKey(data.id),
        domainName: data.name,
        duplicate: data.duplicate,
        primary: data.primary,
        redirectTo: data.redirect_to.name,
      },
    },
  });
}
