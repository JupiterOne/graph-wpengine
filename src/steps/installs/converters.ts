import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { Entities } from '../constants';
import { WpEngineInstall } from '../../types';

export function getInstallKey(id: string): string {
  return `wp_engine_install:${id}`;
}

export function createInstallEntity(data: WpEngineInstall) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: Entities.INSTALL._class,
        _type: Entities.INSTALL._type,
        _key: getInstallKey(data.id),
        name: data.name,
        phpVersion: data.php_version,
        status: data.status,
        cname: data.cname,
        stableIps: data.stable_ips || [],
        environment: data.environment,
        primaryDomain: data.primary_domain,
        isMultistate: data.is_multistate,
      },
    },
  });
}
