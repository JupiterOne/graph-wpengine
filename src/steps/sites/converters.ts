import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { Entities } from '../constants';
import { WpEngineSite } from '../../types';

export function getSiteKey(id: string): string {
  return `wp_engine_site:${id}`;
}

export function createSiteEntity(data: WpEngineSite) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: Entities.SITE._class,
        _type: Entities.SITE._type,
        _key: getSiteKey(data.id),
        name: data.name,
      },
    },
  });
}
