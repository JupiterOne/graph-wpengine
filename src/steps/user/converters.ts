import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { Entities } from '../constants';
import { WpEngineUser } from '../../types';

export function createUserEntity(data: WpEngineUser) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: Entities.USER._class,
        _type: Entities.USER._type,
        // We always want to have "descriptive" keys
        // Also, if later, you need to find users (jobState.findEntity())
        // you'll need to find them by their keys, so you should
        // export and turn this into function similar to getAccountKey
        _key: `wp_engine_user:${data.id}`,
        name: `${data.first_name} ${data.last_name}`,
        username: data.email,
        email: data.email,
      },
    },
  });
}
