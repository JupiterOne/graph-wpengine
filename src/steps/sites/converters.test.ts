import { getMockSite } from '../../../test/mocks';
import { createSiteEntity } from './converters';

describe('#createSiteEntity', () => {
  test('should convert to entity', () => {
    expect(createSiteEntity(getMockSite())).toMatchSnapshot();
  });
});
