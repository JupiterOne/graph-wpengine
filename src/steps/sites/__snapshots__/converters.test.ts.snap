// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createSiteEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Host",
  ],
  "_key": "wp_engine_site:some-id-here",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "account": Object {
          "id": "some-id-here",
        },
        "id": "some-id-here",
        "installs": Array [
          Object {
            "cname": "sample.hostname.com",
            "environment": "sampleEnvironment",
            "id": "some-id-here",
            "is_multisite": false,
            "name": "exampleName",
            "php_version": "1.0",
          },
        ],
        "name": "Doe",
      },
    },
  ],
  "_type": "wp_engine_site",
  "createdOn": undefined,
  "displayName": "Doe",
  "id": "some-id-here",
  "name": "Doe",
}
`;
